tools/sqlite3_api_wrapper/sqlite3_api_wrapper.cpp

```diff
@@ -1588,9 +1587,17 @@ const unsigned char *sqlite3_value_text(sqlite3_value *pVal) {
 	}
 
 	if (pVal->type == SQLiteTypeValue::TEXT || pVal->type == SQLiteTypeValue::BLOB) {
-		// We need don't need to append \0 at the end of string
-		// To know the number of chars or bytes, call function sqlite3_value_bytes()
-		return (const unsigned char *)pVal->str_t.GetDataUnsafe();
+		auto length = pVal->str_t.GetSize();
+		// new string including space for the null-terminated char ('\0')
+		pVal->zMalloc = (char *)malloc(sizeof(char) * length + 1);
```

Comment:
What if `sqlite3_value_text` gets called multiple times on the same value? Wouldn't this leak? Perhaps you could check if zMalloc already exists, and if it exists just return the value in there?

Reply:
Yeah, this cloud happen. I changed the code and added a test.